generic/tests/ping.py

```diff
@@ -5,129 +5,119 @@
 
 from virttest import utils_test
 from virttest import utils_net
-
-
-@error.context_aware
+from virttest import error_context
+
+
+def _ping_with_params(test, params, dest, interface=None,
+                      packet_size=None, interval=None,
+                      count=0, session=None, flood=False):
+    if flood:
+        cmd = "ping " + dest + " -f -q"
+        if interface:
+            cmd += " -S %s" % interface
+        flood_minutes = float(params.get("flood_minutes", 10))
+        status, output = utils_test.raw_ping(cmd, flood_minutes * 60,
+                                             session, logging.debug)
+    else:
+        timeout = float(count) * 1.5
+        status, output = utils_test.ping(dest, count, interval, interface,
+                                         packet_size, session=session,
+                                         timeout=timeout)
+    if status != 0:
+        test.fail("Ping failed, status: %s,"
+                  " output: %s" % (status, output))
+    if params.get("strict_check", "no") == "yes":
+        ratio = utils_test.get_loss_ratio(output)
+        if ratio != 0:
+            test.fail("Loss ratio is %s" % ratio)
+
+
+@error_context.context_aware
 def run(test, params, env):
     """
     Ping the guest with different size of packets.
 
     1) Login to guest
-    2) Ping test on nic(s) from host
+    2) Ping test on nic(s) from host - default_ping/multi_nics
         2.1) Ping with packet size from 0 to 65507
         2.2) Flood ping test
         2.3) Ping test after flood ping, Check if the network is still alive
-    3) Ping test from guest side, packet size is from 0 to 65507
-       (win guest is up to 65500) (Optional)
+    3) Ping test from guest side to external host - ext_host
+        3.1) Ping with packet size from 0 to 65507 (win guest is up to 65500)
+        3.2) Flood ping test
```

Comment:
Seems not have this?

Reply:
@luckyh I wrote it in line113, as you suggested me delete windows flood ping. So this only test ping flood in linux guest and host.

Reply:
Sorry, I misunderstood L113. However, after you explained that, I think it should be `if params.get("os_type") == "linux":`, right?

Reply:
No, we also need test flood ping in default ping part (host ping guest), so it need add  "not ping_ext_host"

Reply:
You mean for windows guests?